File: 3775_reverse_words_with_same_vowel_count/solution.cpp
// LeetCode 3775 - Reverse Words With Same Vowel Count
// https://leetcode.com/problems/reverse-words-with-same-vowel-count/

#include <algorithm>
#include <sstream>
#include <string>
#include <vector>

class Solution {
    static int calc(const std::string& w) {
        int cnt = 0;
        for (char c : w) {
            if (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u') cnt++;
        }
        return cnt;
    }

public:
    std::string reverseWords(std::string s) {
        std::istringstream iss(s);
        std::vector<std::string> words;
        std::string w;
        while (iss >> w) words.push_back(w);

        int cnt = calc(words[0]);
        std::vector<std::string> ans;
        ans.push_back(words[0]);

        for (int i = 1; i < (int)words.size(); i++) {
            w = words[i];
            if (calc(w) == cnt) std::reverse(w.begin(), w.end());
            ans.push_back(w);
        }

        std::ostringstream oss;
        for (int i = 0; i < (int)ans.size(); i++) {
            if (i) oss << ' ';
            oss << ans[i];
        }
        return oss.str();
    }
};

File: 3775_reverse_words_with_same_vowel_count/solution.php
<?php
// LeetCode 3775 - Reverse Words With Same Vowel Count
// https://leetcode.com/problems/reverse-words-with-same-vowel-count/

class Solution {
    function reverseWords($s) {
        $calc = function($w) {
            $cnt = 0;
            $len = strlen($w);
            for ($i = 0; $i < $len; $i++) {
                $c = $w[$i];
                if ($c === 'a' || $c === 'e' || $c === 'i' || $c === 'o' || $c === 'u') $cnt++;
            }
            return $cnt;
        };
        $words = preg_split('/\s+/', trim($s));
        $cnt = $calc($words[0]);
        $ans = $words[0];
        for ($i = 1; $i < count($words); $i++) {
            $w = $words[$i];
            if ($calc($w) === $cnt) $w = strrev($w);
            $ans .= ' ' . $w;
        }
        return $ans;
    }
}
